src/main/java/com/google/cloud/tools/gradle/appengine/flexible/AppEngineFlexiblePlugin.java

```diff
@@ -86,9 +86,8 @@ private void configureExtensions() {
 
           // obtain deploy extension set defaults
           DeployExtension deploy = appengineExtension.getDeploy();
-          if (deploy.getDeployables() == null) {
-            deploy.setDeployables(new File(stageExtension.getStagingDirectory(), "app.yaml"));
-          }
+          deploy.updateDeployables(new File(stageExtension.getStagingDirectory(), "app.yaml"));
```

Comment:
Since this is no longer user-configurable, maybe it makes sense to have it just be on the `DeployAllTask` as a setter.

Reply:
If I did this, I'd still need to set it in the deploy extension somehow. It extends DeployConfiguration, which means I need a `getDeployables()`.

Reply:
Yeah, its kind of a weird consequence of how I wrote it initially. DeployExtension doesn't seem to map nicely anymore.

One option is not even have the setter, and have some kind of weird copy constructor that generates a DeployExtension with deployables being a immutable member of DeployExtension.

```
private final List<File> deployables;

...

DeployExtension(DeployExtension deploy, List<File> deployables) {
  // copy stuff as done in the other copy constructor
   this.deployables = ImmutableList.of(deployables)
}
```

remove the setter (or updater), and call this constructor in the task directly (Deploy, DeployAll)?

Reply:
and then you'd probably, as Q suggested, pass in the necessary info to the task directly.